### ISSUE TITLE
Obsidian Attendance Detail block doesn't create attendance records for people who didn't attend

### ISSUE THREAD

**Issue by leahjennings** [SparkDevNetwork/Rock#5554]
### Please go through all the tasks below

- [X] Check this box only after you have successfully completed both the above tasks

### Please provide a brief description of the problem. Please do not forget to attach the relevant screenshots from your side.

Since our upgrade to v15, we've noticed that the _Percent Attended_ column of the Group Attendance page _always_ shows 100%, regardless of the number of people who actually attended. Here's a screenshot of what it looks like in our instance:
<img width="1728" alt="image" src="https://github.com/SparkDevNetwork/Rock/assets/66277785/53002db4-cdc7-42a2-8abf-a7b454c8662c">

Looking at [how _AttendanceRate_ is calculated](https://github.com/SparkDevNetwork/Rock/blob/882c1e975a38b00902dcde068b5187dd418e961c/Rock/Model/Event/AttendanceOccurrence/AttendanceOccurrence.Logic.cs#L67), it appears that it is taking the total number of attendees with the "Did Attend" flag set to true divided by the total number of attendee records for that occurrence. Looking through the table, the people who didn't attend don't have attendance records with "Did Attend" set to false, therefore the math will always calculate to 100% attended.

I used the SQL below to determine this behavior is incorrect. Looking at the same group in our production instance that is represented in the screenshot above, before the upgrade to v15 there was an attendance record created for each group member, with the Did Attend flag set to true if they were marked as attended. Post-v15 upgrade, only the people who were marked present have attendance records created.
```
SELECT *
  FROM Attendance a
    JOIN AttendanceOccurrence ao ON (ao.Id = a.OccurrenceId)
 WHERE ao.GroupId = ### -- ### is the Group Id of the group in question
 ORDER BY ao.Id DESC
```

The post-v15 upgrade is significant because ever since we have been using the Obsidian Attendance Detail block.

### Expected Behavior

I would expect that attendance records would get created for all individuals in the group, with the Did Attend flag correctly set to true/false depending on whether or not they were marked as attended.

### Actual Behavior

Attendance records only get created for those who attended, so the Percent Attended column on the group attendance page is always 100%.

### Steps to Reproduce

1. On the demo site, hover over the Person icon in the sidebar and click on _Group Viewer_
2. Expand _Section A_, and click on one of the groups in the list (for this example, I'm using _Decker Group_)
3. Click on the attendance checkbox icon
![image](https://github.com/SparkDevNetwork/Rock/assets/66277785/3585c262-671e-4fac-bfaf-60214f0127ea)

4. Click on one of the attendance occurrences (I clicked on 8/14/23)
![image](https://github.com/SparkDevNetwork/Rock/assets/66277785/769b7e3c-b88f-4237-a4b5-05dbaaf1d2f2)

5. Click on Cindy Decker and Ted Decker to mark them as "attended"
   - Note that this _should_ calculate percent attended at 50%, since only half of the people were marked attended
6. Click on the Back button
7. Notice that 8/14/23 occurrence now shows 100% attended even though only 2 of the 4 were marked attended
![image](https://github.com/SparkDevNetwork/Rock/assets/66277785/40c02fe5-f534-4ce4-9ae2-ea37cac3745b)

### Rock Version

Rock McKinley 15.0 (1.15.0.16)

### Client Culture Setting

en-US

**Comment by nairdo** [comment]
Hi @leahjennings  We think this is possibly a duplicate of #5473 which was fixed in v15.1 via https://github.com/SparkDevNetwork/Rock/commit/ddbcf61ba9dd0787396cf31d3c3c32b42819a895

I updated the Demo site to v15.1 and tested this issue there to verify, but please let me know if I'm mistaken.
![chrome_J2GW1ixw1F](https://github.com/SparkDevNetwork/Rock/assets/383807/c07f6e4b-1dab-49b2-bd37-6a3384941c90)

**Comment by leahjennings** [comment]
Agreed @nairdo - thanks for the clarity! Apologies for the duplicate!
